streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorNodeTest.java

```diff
@@ -34,36 +40,106 @@
+
+        final InternalProcessorContext<Object, Object> internalProcessorContext = mockInternalProcessorContext();
+        node.setProcessingExceptionHandler(new ProcessingExceptionHandlerMock(ProcessingExceptionHandler.ProcessingHandlerResponse.FAIL, internalProcessorContext));
+        node.init(internalProcessorContext);
+
+        final FailedProcessingException failedProcessingException = assertThrows(FailedProcessingException.class,
+            () -> node.process(new Record<>(KEY, VALUE, TIMESTAMP)));
+
+        assertEquals("java.lang.RuntimeException: Processing exception should be caught and handled by the processing exception handler.",
+            failedProcessingException.getMessage());
+
+        assertTrue(failedProcessingException.getCause() instanceof RuntimeException);
+        assertEquals("Processing exception should be caught and handled by the processing exception handler.",
+            failedProcessingException.getCause().getMessage());
+    }
+
+    @Test
+    public void shouldNotThrowStreamsExceptionWhenProcessingMarkedAsContinue() {
+        final ProcessorNode<Object, Object, Object, Object> node =
+            new ProcessorNode<>(NAME, new ProcessingExceptionProcessor(), Collections.emptySet());
+
+        final InternalProcessorContext<Object, Object> internalProcessorContext = mockInternalProcessorContext();
+        node.setProcessingExceptionHandler(new ProcessingExceptionHandlerMock(ProcessingExceptionHandler.ProcessingHandlerResponse.CONTINUE, internalProcessorContext));
+        node.init(internalProcessorContext);
+
+        assertDoesNotThrow(() -> node.process(new Record<>(KEY, VALUE, TIMESTAMP)));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldNotHandleInternalFailedProcessingException() {
+        final ProcessingExceptionHandler processingExceptionHandler = mock(ProcessingExceptionHandler.class);
+
+        final ProcessorNode<Object, Object, Object, Object> node =
+            new ProcessorNode<>(NAME, new FailedProcessingExceptionProcessor(), Collections.emptySet());
+        node.setProcessingExceptionHandler(processingExceptionHandler);
+
+        final InternalProcessorContext<Object, Object> internalProcessorContext = mock(InternalProcessorContext.class);
+        when(internalProcessorContext.taskId()).thenReturn(TASK_ID);
+        when(internalProcessorContext.metrics()).thenReturn(new StreamsMetricsImpl(new Metrics(), "test-client", StreamsConfig.METRICS_LATEST, new MockTime()));
+        node.init(internalProcessorContext);
+
+        final FailedProcessingException failedProcessingException = assertThrows(FailedProcessingException.class,
+            () -> node.process(new Record<>(KEY, VALUE, TIMESTAMP)));
+
+        assertEquals("java.lang.RuntimeException: FailedProcessingException should not be caught and handled by the processing exception handler.",
+            failedProcessingException.getMessage());
```

Comment:
I would prefer to verify the cause and the message of the cause. I am not sure the format of the message when the exception does not have its own message but uses the message of the cause is standardized in Java. Different java implementations might have slightly different formats that would fail the test.

Reply:
I updated this test to a parameterized test that checks the cause and the cause message for a given exception to ignore. Let me know if you find it acceptable